File: src/com/company/sort/BubbleSort.java
package com.company.sort;

public class BubbleSort {

    /*
        64, 25, 12, 22, 11
        25 64 12 22 11
        25 12 64 22 11
        25 12 22 64 11
        25 12 22 11 64
        ===========
        12 25 22 11 64
        12 22 25 11 64
        12 22 11 25 64
        ===========
        12 22 11 25 64
        12 11 22 25 64
        ===========
        11 12 22 25 64
        ===========
        Sorted array
        11 12 22 25 64

     */


    public static void main(String[] args) {
        BubbleSort ob = new BubbleSort();
        int arr[] = {64, 25, 12, 22, 11};
        ob.sort(arr);
        System.out.println("Sorted array");
        ob.printArray(arr);
    }

    private void sort(int[] arr) {
        for (int i = 0; i < arr.length - 1; i++) {
            for (int j = 0; j < arr.length - 1 - i; j++) {
                if (arr[j] > arr[j + 1]) {
                    int temp = arr[j];
                    arr[j] = arr[j + 1];
                    arr[j + 1] = temp;
                }
                printArray(arr);
            }
            System.out.println("===========");
        }

    }

    private void printArray(int[] arr) {
        for (int i1 : arr) {
            System.out.print(i1 + " ");
        }
        System.out.println();
    }


}

File: src/com/company/gfg/sorting/BubbleSort.kt
package com.company.gfg.sorting

import java.util.*


fun main() {
    println(bubbleSortKotlin(arrayListOf(3, 44, 38, 5, 47, 15, 36)))
}

fun sort(arr: IntArray) {
    for (i in 0 until arr.size - 1) {
        for (j in 0 until arr.size - 1 - i) {
            if (arr[j] > arr[j + 1]) {
                val temp = arr[j]
                arr[j] = arr[j + 1]
                arr[j + 1] = temp
            }
        }
        println("===========")
    }
}

fun bubbleSortKotlin(arr: ArrayList<Int>): Int {
    var left: Int
    var right: Int
    var swap = true

    while (swap) {
        swap = false
        arr.withIndex().forEach { (index, it) ->
            if (index != arr.size - 1) {
                left = it
                right = arr[index + 1]
                if (left > right) {
                    val temp = left
                    arr[index] = right
                    arr[index + 1] = temp
                    swap = true
                }
            }
        }
    }

    arr.forEach {
        println(it)
    }
    return 0
}

